### ISSUE TITLE
panel can't save setting when "Apply to metrics" is undefined

### ISSUE THREAD

**Issue by dennybo** [algenty/grafana-flowcharting#370]
![](https://daniel-markdown.oss-cn-shanghai.aliyuncs.com/pictures/rJ5mGBVDnb.png?versionId=CAEQPhiBgIC0zcTHmBgiIDZkZTFlYmZmZGQwYTRmNDJhMzVhYTRlM2Q3YTZmOGEz)

**Comment by EwaldN** [comment]
Yes i have exacly the same problem. When i have "undefined" selected and close and reopen the panel all the options settings for all rules vanished. Does anyone has a advice how to solve the issue?

**Comment by Scavallarin** [comment]
For me the same, is full of bugs, such a shame.

**Comment by romainlaisne** [comment]
I experience the same. It seems this happens only while using Grafana transformations.

**Comment by ftoledo** [comment]
same on grafana 9.3.2 and zabbix data source.
any workaround?

**Comment by kranthikirang** [comment]
this is because grafana drops refId after transformation. https://github.com/grafana/grafana/issues/56312

**Comment by Thuaiqian** [comment]
You can try to resolve this issue by selecting the option to disable data alignment.

![企业微信截图_16735061716278](https://user-images.githubusercontent.com/92358460/212000029-1f2e20e9-04b0-41b3-b8a7-bd07658273a1.png)

**Comment by romainlaisne** [comment]
> You can try to resolve this issue by selecting the option to disable data alignment.
> 
> ![企业微信截图_16735061716278](https://user-images.githubusercontent.com/92358460/212000029-1f2e20e9-04b0-41b3-b8a7-bd07658273a1.png)

I believe this example is for Zabbix plugin, not the Flowcharting plugin. Right?
I don't see this option in the Flowcharting plugin.

**Comment by romainlaisne** [comment]
The solution here is probably to update Grafana:
https://github.com/grafana/grafana/issues/56312

**Comment by ftoledo** [comment]
> The solution here is probably to update Grafana: [grafana/grafana#56312](https://github.com/grafana/grafana/issues/56312)

upgrade grafana to 9.3.6 still have this issue.
